### DISCUSSION TITLE
AlpineJs' x-sort directive not working at all

### DISCUSSION THREAD

**Discussion by Justin-del** [alpinejs/alpine#4255] [1. Help] [answered]
I have the following simple code.
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
    <!-- Alpine Plugins -->
<script defer src="https://cdn.jsdelivr.net/npm/@alpinejs/sort@3.x.x/dist/cdn.min.js"></script>
 
<!-- Alpine Core -->
<script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js"></script>
</head>
<body>
    <ul x-sort>
        <li x-sort:item>foo</li>
        <li x-sort:item>bar</li>
        <li x-sort:item>baz</li>
    </ul>
</body>
</html>
```

I can't drag any of the list items when I run the above code using Visual Studio Code's live server extension. What am I doing wrong?

**Comment by Saeven** [comment]
I've got a working BIN here if it helps you:
https://jsbin.com/nazucit/2/edit?html,output

**Comment by SimoTod** [comment, marked as answer]
You don't have any x-data

**Reply by Justin-del** [reply]
Thanks, Simo Tod. It is funny that in the original x-sort documentation at https://alpinejs.dev/plugins/sort, it didn't say that I need that directive in order to use x-sort.

**Reply by SimoTod** [reply]
It's in the paragraph about starting with Alpine. You always need x-data on a parent tag (or on the same tag) for any directive.

**Reply by Justin-del** [reply]
Oh, I see.  Thanks, SimoTod.
